src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -26,24 +26,10 @@ public class QnAService {
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
         Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        List<Answer> answers = question.getAnswers();
-        for (Answer answer : answers) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : answers) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+
+        question.delete(loginUser);
+        DeleteHistoryTargets deleteHistoryTargets = question.asDeleteHistoryTargets(loginUser);
```

Comment:
> "메서드는 한가지 역할만 해야 한다.(SRP)"를 지키기 위해 삭제 로직에 대해, 책임(역할)을 아래와 같이 나누었습니다. 리뷰어님은 이 부분은 합치는게 좋다고 생각하실까요?

> delete() - 삭제 대상인 Question과 Answer의 상태를 삭제로 변경하는 역할
asDeleteHistoryTargets() 실제 DB 삭제를 위해, 삭제 리스트를 받는 역할 (실제 삭제는 Service에서 하기 때문에, 역할을 나눔)

삭제를 하면 히스토리가 생성되는 것이 하나의 책임이 아닐까요?
책임을 두가지의 행위로 구분지어서 생각하신것 같아요!

1. 삭제한다.
2. 히스토리를 생성한다.

위의 2가지 행위는 책임을 다하기 위한 기능이라고 생각합니다.

삭제하면 항상 히스토리가 쌓이기에 리턴값으로 history 받는게 자연스러울것 같아요!

```suggestion
        DeleteHistoryTargets history = question.delete(loginUser);
```

히스토리 = 질문을 삭제한다(로그인 유저가)

Reply:
이 부분은 제가 역할에 혼동이 있는 듯 했습니다.
하나로 합치는 것으로 변경 했습니다!